pkg/applications/helmclient/client.go

```diff
@@ -286,6 +286,9 @@ func (h HelmClient) Upgrade(chartLoc, releaseName string, values map[string]inte
 	upgradeClient.Atomic = deployOpts.atomic
 	upgradeClient.EnableDNS = deployOpts.enableDNS
 
+	// restrict history to avoid OOM kill
+	upgradeClient.MaxHistory = 1
```

Comment:
I would rather Keep a couple of Histories around. In case something breaks we have an easier time debugging maybe?

Reply:
If the upgrade fails, the controller will requeue the event, so history will quickly reach the limit. However, I do some additional testing, and when Helm purges the history, it keeps the last successful release version.

For example, this is the history with a limit of 1

- **revision 1**: successful install  
- **revision 2**: fail upgrade
- **revision 3**: fail upgrade 

```
REVISION	UPDATED                 	STATUS  	CHART             	APP VERSION	DESCRIPTION
1       	Fri Mar 24 11:16:29 2023	deployed	examplechart-0.2.0	           	Install complete
3       	Fri Mar 24 11:17:14 2023	failed  	examplechart-0.2.0	           	Upgrade "testchart" failed: timed out waiting for the condition
```

The **revision 2** has been deleted when history has been purged but **revision 1** (last sucessful has been kept)

@WeirdMachine  So it should be enough for debugging, isn't it?
Anyway, I should update the test to show this behavior...

Reply:
Ah cool I did not know about this. Thanks for the explanation :) 
I think simply change the comment above to briefly explain this should be fine